### PR TITLE
feat(hooks): create useTourFlow hook with comprehensive tests

### PR DESCRIPTION
**Pull request by aafre**
Create Layer 3 hook for managing tour and idle nudge flows:

Hook Features:
- Initial tour launch with 1.5s delay for new users
- Tour re-launch after OAuth sign-in and migration completion
- Idle nudge tooltip after 5 minutes for anonymous users
- Infinite loop prevention with refs (hasLaunchedTourAfterSignIn, justSignedInFromTour)
- Sign-in from tour tracking with handleSignInFromTour/handleSignInSuccess

Implementation Details:
- Dependency injection pattern with props from contexts and hooks
- Three separate effects: initial tour, re-launch, and idle nudge
- Comprehensive guard conditions to prevent re-launch loops
- Auto-dismiss idle tooltip after 10 seconds
- Console logging for debugging tour re-launch

Test Coverage (99.21% statements, 96.77% branches, 100% functions):
- 25 tests covering all flows and edge cases
- Fake timers for testing async behavior
- Full simulation of tour → sign-in → migration → re-launch flow
- Tests for all guard conditions and error scenarios

Task: TASK-021, TASK-022

### CODE REVIEW

File: resume-builder-ui/src/hooks/editor/useTourFlow.ts [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+
+        // Auto-dismiss after 10 seconds
+        setTimeout(() => setShowIdleTooltip(false), 10000);
+      }, 5 * 60 * 1000); // 5 minutes
+
+      return () => {
+        if (idleTimerRef.current) clearTimeout(idleTimerRef.current);
+      };
+    }
+  }, [isAnonymous, hasShownIdleNudge, authLoading, markIdleNudgeShown]);
```
[the remark this answers is not in the document]
Comment (aafre) [reply resume-builder-ui/src/hooks/editor/useTourFlow.ts]: @gemini-code-assist - fixed, review now.
